### ISSUE TITLE
Does this train from scratch or uses pre-trained weights?

### ISSUE THREAD

**Issue by Y-T-G** [naseemap47/YOLO-NAS#35]
I saw the `weights` flag in the code to specify pre-trained weights, but I don't see that argument being utilized anywhere in the training code. So is it currently training from scratch always?

**Comment by naseemap47** [comment]
Hi @Y-T-G ,
It will NOT train from scrach.
`weights` flag is for giving **pre-trained weight**. by default its `coco`,
So it will train on **coco weight**
checkout https://github.com/naseemap47/YOLO-NAS/blob/master/train.py#L129

I think this will help you to understand.
If you  have any questions, please let me know.

**Comment by Y-T-G** [comment]
@naseemap47 I see. What do I do if I want to train from scratch?

**Comment by naseemap47** [comment]
Hi @Y-T-G ,
I didn't find any option for that.
Checkout **super-gradients** repo: https://github.com/Deci-AI/super-gradients
If you find any, please let me know.

**Comment by Y-T-G** [comment]
@naseemap47 I think for that you just don't pass an argument to the pretrained_weights as it says in the comment here[](https://github.com/Deci-AI/super-gradients/blob/f49a35811638905bbe9b19d4521aa39891315d43/documentation/source/Segmentation.md?plain=1#L60) .

**Comment by naseemap47** [comment]
Hi @Y-T-G ,
Thank you for sharing. I will look into it, and I will update this ASAP.
Thank you.

**Comment by naseemap47** [comment]
Hi @Y-T-G ,
I think `pretrained_weights="cityscapes" ` option is removed for detection.
When I tried its showing unable to get `cityscapes` model.
Did you face any issue ?

**Comment by Y-T-G** [comment]
@naseemap47 Not sure about that. What I meant was you can remove that argument altogether from the function call, i.e. just don't pass `pretrained_weights` and it should train from scratch. You can verify that from the starting mAP which will be close to 0 because it's training from scratch.

**Comment by Y-T-G** [comment]
https://github.com/Y-T-G/YOLO-NAS/commit/ed597f270994471fe5aec12fd5a38809d1f35e13

This should work too. To train from scratch simply use `--weights=""`.

**Comment by naseemap47** [comment]
Hi @Y-T-G ,
Thank you for finding this.
I will let you know after checking this.
